devlib/instrument/__init__.py

```diff
@@ -205,16 +205,19 @@ def reset(self, sites=None, kinds=None, channels=None):
             if isinstance(kinds, basestring):
                 kinds = [kinds]
             self.active_channels = []
-            for chan_name in (channels or []):
-                try:
-                    self.active_channels.append(self.channels[chan_name])
-                except KeyError:
-                    msg = 'Unexpected channel "{}"; must be in {}'
-                    raise ValueError(msg.format(chan_name, self.channels.keys()))
-            for chan in self.channels.values():
-                if (kinds is None or chan.kind in kinds) and \
-                   (sites is None or chan.site in sites):
-                    self.active_channels.append(chan)
+            if channels is not None:
+                for chan_name in channels:
+                    print chan_name
+                    try:
+                        self.active_channels.append(self.channels[chan_name])
+                    except KeyError:
+                        msg = 'Unexpected channel "{}"; must be in {}'
+                        raise ValueError(msg.format(chan_name, self.channels.keys()))
+            else:
+                for chan in self.channels.values():
```

Comment:
in the `else` clause, `channels is None`` so this will raise a `NoAttributeError`